packages/router/src/grpc/view-protocol-server/broadcast-transaction.test.ts

```diff
@@ -0,0 +1,483 @@
+    };
+
+    mockIndexedDb = {
+      subscribe: (table: string) => {
+        if (table === 'TRANSACTION_INFO') return mockTransactionInfoSubscription;
+        throw new Error('Table not supported');
+      },
+    };
+    mockServices = {
+      getWalletServices: vi.fn(() => Promise.resolve({ indexedDb: mockIndexedDb })),
+      querier: {
+        tendermint: mockTendermint,
+      },
+    };
+
+    mockCtx = createHandlerContext({
+      service: ViewProtocolService,
+      method: ViewProtocolService.methods.broadcastTransaction,
+      protocolName: 'mock',
+      requestMethod: 'MOCK',
+      contextValues: createContextValues().set(servicesCtx, mockServices as unknown as Services),
+    });
+
+    broadcastTransactionRequest = new BroadcastTransactionRequest({
+      transaction: transactionData,
+    });
+  });
+
+  test('should successfully broadcastTransaction without await detection', async () => {
+    mockTendermint.broadcastTx.mockResolvedValue(transactionIdData);
+
+    const broadcastResponses: BroadcastTransactionResponse[] = [];
+    for await (const response of broadcastTransaction(broadcastTransactionRequest, mockCtx)) {
+      broadcastResponses.push(new BroadcastTransactionResponse(response));
+    }
+    expect(broadcastResponses.length === 1).toBeTruthy();
+    expect(broadcastResponses[0]?.status.case === 'broadcastSuccess').toBeTruthy();
+  });
+
+  test('should successfully broadcastTransaction with await detection', async () => {
+    const detectionHeight = 222n;
+    const txInfo = new TransactionInfo({
+      transaction: transactionData,
+      height: detectionHeight,
+      id: transactionIdData,
+    });
+
+    mockTendermint.broadcastTx.mockResolvedValue(transactionIdData);
+    txSubNext.mockResolvedValueOnce({
+      value: { value: txInfo.toJson(), table: 'TRANSACTION_INFO' },
+    });
+
+    broadcastTransactionRequest.awaitDetection = true;
+
+    const broadcastResponses: BroadcastTransactionResponse[] = [];
+    for await (const response of broadcastTransaction(broadcastTransactionRequest, mockCtx)) {
+      broadcastResponses.push(new BroadcastTransactionResponse(response));
+    }
+    expect(broadcastResponses.length === 2).toBeTruthy();
+    expect(broadcastResponses[0]?.status.case === 'broadcastSuccess').toBeTruthy();
+    expect(broadcastResponses[1]?.status.case === 'confirmed').toBeTruthy();
+    expect(broadcastResponses[1]?.status.value?.id?.equals(transactionIdData)).toBeTruthy();
+  });
+
+  test('should throw error if broadcast transaction id disagrees', async () => {
+    mockTendermint.broadcastTx.mockResolvedValue(new TransactionId());
+    await expect(
+      (async () => {
+        // eslint-disable-next-line @typescript-eslint/no-unused-vars, no-empty
+        for await (const _ of broadcastTransaction(broadcastTransactionRequest, mockCtx)); // eslint-disable-line   no-empty
+      })(),
+    ).rejects.toThrow('broadcast transaction id disagrees');
```

Comment:
I haven't found a better way to verify that async generator throw an error

Reply:
Another way is to do this:

```typescript
  test('shsould throw error if broadcast transaction id disagrees', async () => {
    mockTendermint.broadcastTx.mockResolvedValue(new TransactionId());

    try {
      for await (const _ of broadcastTransaction(broadcastTransactionRequest, mockCtx)) {}
      throw new Error('broadcastTransaction did not throw an error as expected');
    } catch (error) {
      expect((error as ConnectError).message).contains('broadcast transaction id disagrees');
    }
  });
```

But don't feel strongly